packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.controller.js

```diff
@@ -1,56 +1,109 @@
 import filter from 'lodash/filter';
 import includes from 'lodash/includes';
+import get from "lodash/get";
 
 export default class {
   /* @ngInject */
-  constructor($filter, $timeout) {
+  constructor($scope, $filter, $timeout, $translate, HostingCdnOrderService) {
+    this.$scope = $scope;
     this.$filter = $filter;
     this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.HostingCdnOrderService = HostingCdnOrderService;
+    this.cdnCase = null;
+    this.cdn1Price = null;
   }
 
   $onInit() {
     // Auto-select duration
+    console.log('ZM:: serviceInfo', this.serviceInfo);
+    console.log('ZM:: catalogAddon', this.catalogAddon);
+    console.log('ZM:: catalogAddons', this.catalogAddons);
     this.prices = filter(this.catalogAddon.pricings, ({ capacities }) =>
       includes(capacities, 'renew'),
     );
     [this.price] = this.prices;
     this.interval = this.price.interval;
     this.isEditable = true;
+    console.log('ZM:: price', this.price);
 
     if (this.catalogAddon.pricings.length === 1) {
       // Go directly to the next step
       this.currentIndex = 1;
       this.isEditable = false;
     }
+
+    if (this.isIncludedCDN) {
+      this.cdnCase = 'included';
+    } else if (this.isPayableCDN) {
+      this.cdnCase = 'payable';
+    } else if (!this.hasCDN) {
+      this.cdnCase = 'without';
+    }
+
+    const cdn1Addon = this.catalogAddons.addons.find(addon => addon.planCode === "cdn_business");
+    const cdn1Prices = filter(cdn1Addon.pricings, ({ capacities }) =>
+      includes(capacities, 'renew'),
+    );
+    [this.cdn1Price] = cdn1Prices;
+
+    console.log('ZM:: cdn1Price', cdn1Addon, this.cdn1Price);
+    console.log('ZM:: cdnCase', this.cdnCase);
   }
 
   resetCart() {
     if (this.cart) {
       this.cart = undefined;
       this.cartId = undefined;
+      this.addonPlan = undefined;
+      this.serviceId = undefined;
     }
   }
 
   async prepareCheckout() {
```

Comment:
Not yours, but I'm not sure this is handled correctly by angularJS digest cycle. maybe we should consider using Promise for now

Reply:
no changes